.github/workflows/mcp_unit_test.yml

```diff
@@ -0,0 +1,58 @@
+name: 'MCP Unit Tests'
+
+on:
+  push:
+    branches:
+    - master
```

Comment:
It seems that you cannot specify to run job based on branch that you are attempting to merge. 
Can filter branches based on pattern/prefix, but that would be a job when a push/pr happens to that branch ([docs](https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestbranchestags)).
Might be able to make a step/job that gets the name of the source branch for the pr using `github.head_ref`, and then do something conditional like '[job if](https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idif) the branch had a the prefix `feat_MCP`'
Could also see if it is possible to only run tests if changed files were within `terraform-infrastructure-modules/mcp` or `terraform-infrastructure-modules/tests/mcp`

Reply:
> Might be able to make a step/job that gets the name of the source branch for the pr using github.head_ref, and then do something conditional like 'job if the branch had a the prefix feat_MCP'

go with this

Reply:
Have added conditional so that the job only runs if the calling branch for pr or push contains "mcp". There will be a "skipped" result in actions tab if the pr/push was from a branch not containing mcp. 
The action now also runs on pr for master, as well as push to master